### PR TITLE
Ensure tables with different gc grace seconds mismatch and hence will be updated

### PR DESCRIPTION
**Pull request by hsaraogi**
**Goals (and why)**: Unit test for CfDef mismatch when tables have different GC grace seconds. liked to #1929.

**Implementation Description (bullets)**: test

**Concerns (what feedback would you like?)**: na

**Where should we start reviewing?**: small PR

**Priority (whenever / two weeks / yesterday)**: whenever

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2017)

### CODE REVIEW

File: atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/ColumnFamilyDefinitionsTest.java [hunk 1]
```diff
@@ -39,4 +43,20 @@ public void compactionStrategiesShouldMatchWithOrWithoutPackageName() {
...
                 ColumnFamilyDefinitions.isMatchingCf(fullyQualified, onlyClassName));
     }
+
+
+    @Test
+    public void ifCfDefsHaveDifferentGcGraceSecondsIsMatchingCfReturnsFalse() {
+        CfDef clientSideTable = ColumnFamilyDefinitions.getCfDef(
+                "test_keyspace",
+                TableReference.fromString("test_table"),
+                AtlasDbConstants.GENERIC_TABLE_METADATA);
```
Comment (gmaretic) [inline atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/ColumnFamilyDefinitionsTest.java]: I wouldn't rely on defaults, e.g., `setGc_grace_seconds(ONE_DAY / 2);`?

Comment (hsaraogi) [inline atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/ColumnFamilyDefinitionsTest.java]: The aim here is to check if default tables are updates from the ONE_DAY setting.. renaming the test.

Comment (gmaretic) [inline atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/ColumnFamilyDefinitionsTest.java]: Ah, then I would rename the constant to something like `OLD_DEFAULT_VALUE`, but I'm fine with this as well

File: atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/ColumnFamilyDefinitionsTest.java [hunk 2]
```diff
@@ -15,14 +15,18 @@
...
 import static org.junit.Assert.assertTrue;
 
 import org.apache.cassandra.thrift.CfDef;
 import org.junit.Test;
 
+import com.palantir.atlasdb.AtlasDbConstants;
 import com.palantir.atlasdb.keyvalue.api.TableReference;
 
 public class ColumnFamilyDefinitionsTest {
+    private static final int ONE_DAY = 4 * 24 * 60 * 60;
```
Comment (tpetracca) [inline atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/ColumnFamilyDefinitionsTest.java]: do you mean 4 days? also should add units to the variable `FOUR_DAYS_IN_SECONDS`

File: atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/ColumnFamilyDefinitionsTest.java [hunk 3]
```diff
@@ -39,4 +43,20 @@ public void compactionStrategiesShouldMatchWithOrWithoutPackageName() {
                         onlyClassName.compaction_strategy),
                 ColumnFamilyDefinitions.isMatchingCf(fullyQualified, onlyClassName));
     }
+
+
+    @Test
+    public void cfDefWithOldAndNewDefaultValuesShouldNotMatch() {
```
Comment (tpetracca) [inline atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/ColumnFamilyDefinitionsTest.java]: Maybe rename this test to `cfDefWithDifferingGcGraceSecondsValuesShouldNotMatch`. I think there's a world where we one day distinguish between different values of the cfDef being considered in the matching or not so this test should be clear that it's testing `gc_grace_seconds`

File: atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/ColumnFamilyDefinitionsTest.java (hunk 2 above)
Comment (hsaraogi) [inline atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/ColumnFamilyDefinitionsTest.java]: Oops missed that.

File: atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/ColumnFamilyDefinitionsTest.java (hunk 1 above)
Comment (tpetracca) [inline atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/ColumnFamilyDefinitionsTest.java]: yea if we change the default value to something that equals the value in this test (which we likely will not) the test will fail and someone will just have to change the test value, so it's probably fine as is. But I see your point.

File: atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/ColumnFamilyDefinitionsTest.java [hunk 4]
```diff
@@ -39,4 +43,20 @@ public void compactionStrategiesShouldMatchWithOrWithoutPackageName() {
                         onlyClassName.compaction_strategy),
                 ColumnFamilyDefinitions.isMatchingCf(fullyQualified, onlyClassName));
     }
+
+
```
Comment (ashrayjain) [inline atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/ColumnFamilyDefinitionsTest.java]: super duper nit: extra newline :D

**Pull request merged**